ui/src/pages/motherPostNatalCare.jsx

```diff
@@ -0,0 +1,429 @@
+  import FormFields from "../components/FormFields";
+  import postnatalCareFields from "../lib/forms/motherPostNatalCare";
+
+export default function MotherPostnatalCare({ userData }) {
+    let [patient, setPatient] = useState({});
+    let [visit, setVisit] = useState();
+    let [open, setOpen] = useState(false);
+    let [loading, setLoading] = useState(false);
+    let [message, setMessage] = useState(false);
+    let [observations, setObservations] = useState([]);
+    let isMobile = useMediaQuery("(max-width:600px)");
+    const [newVisit, setNewVisit] = useState(false);
+    let [motherPostNatalCare, setmotherPostNatalCare] = useState({});
+    let [motherPostNatalCareEncounters, setmotherPostNatalCareEncounters] = useState(
+        []
+    );
+    const handleClose = () => setOpenModal(false);
+    const handleOpen = () => setOpenModal(true);
+    const [value, setValue] = useState("1");
+    let [openModal, setOpenModal] = useState(false);
+
+    const [inputData, setInputData] = useState({});
+    const [preview, setPreview] = useState(false);
+
+
+    const navigate = useNavigate();
+    
+    const fieldValues = Object.values(postnatalCareFields).flat();
+    const validationFields = fieldValues.map((item) => ({
+        [item.name]: item.validate,
+    }));
+
+    const validationSchema = yup.object({
+        ...Object.assign({}, ...validationFields),
+    });
+
+    const initialValues = Object.assign(
+        {},
+        ...fieldValues.map((item) => ({ [item.name]: "" }))
+    );
+
+    const formik = useFormik({
+        initialValues: {
+            ...initialValues,
+            facilityName: userData?.facilityName,
+            kmhflCode: userData?.kmhflCode,
+        },
+        validationSchema: validationSchema,
+        onSubmit: (values) => {
+            console.log(values);
+            setPreview(true);
+            setInputData(values);
+        },
+    });
+
+    function prompt(text) {
+        setMessage(text);
+        setOpen(true);
+        setTimeout(() => setOpen(false), 4000);
+    }
+    const handleChange = (event, newValue) => setValue(newValue);
+
+    useEffect(() => {
+        let visit = window.localStorage.getItem("currentPatient");
+        if (!visit) {
+          setMessage(
+            "No patient visit not been initiated. To start a visit, Select a patient in the Patients list"
+          );
+          setOpen(true);
+          setTimeout(() => {
+            setOpen(false);
+          }, 4000);
+          return;
+        }
+        setVisit(JSON.parse(visit));
+        return;
+      }, []);
+    
+      useEffect(() => {
+        let visit = window.localStorage.getItem("currentPatient");
+        if (!visit) {
+          prompt(
+            "No patient visit not been initiated. To start a visit, Select a patient in the Patients list"
+          );
+          return;
+        }
+        setVisit(JSON.parse(visit));
+        return;
+      }, []);
+    
+      useEffect(() => {
```

Comment:
Only get the encounters in this hook. Remove the currentPatient item retrieval from the local storage as it has already been retrieved in the first useEffect hook and its state set to the visit variable.